pkg/kubelet/kubelet.go

```diff
@@ -287,24 +287,19 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
 }
 
 // PreInitRuntimeService will init runtime service before RunKubelet.
-func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
-	kubeDeps *Dependencies,
-	remoteRuntimeEndpoint string,
-	remoteImageEndpoint string) error {
-	// remoteImageEndpoint is same as remoteRuntimeEndpoint if not explicitly specified
-	if remoteRuntimeEndpoint != "" && remoteImageEndpoint == "" {
-		remoteImageEndpoint = remoteRuntimeEndpoint
+func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration, kubeDeps *Dependencies) error {
+	if kubeCfg.ContainerRuntimeEndpoint != "" && kubeCfg.ImageServiceEndpoint == "" {
+		kubeCfg.ImageServiceEndpoint = kubeCfg.ContainerRuntimeEndpoint
 	}
```

Comment:
prefer avoiding mutating kubeCfg by declaring and defaulting a local remoteImageEndpoint variable and continuing to pass that to the functions below

Reply:
Updated to avoid mutating kubeCfg.